hrrecruitment/CRM/HRRecruitment/Page/WidgetJs.php

```diff
@@ -117,7 +117,13 @@ public static function vacancyListDisplay() {
       $row[$vacancyVal['id']]['id'] = $vacancyVal['id'];
       $row[$vacancyVal['id']]['position'] = CRM_Utils_Array::value('position', $vacancyVal);
       $row[$vacancyVal['id']]['positionLink'] = "{$base_url}/civicrm/vacancy/info?id={$vacancyVal['id']}";
-      $row[$vacancyVal['id']]['location'] = CRM_Utils_Array::value('location', $vacancyVal);
+
+      $locationValue = CRM_Utils_Array::value('location', $vacancyVal);
+      if ($locationValue !== FALSE) {
+        $locationValue = CRM_Core_PseudoConstant::getLabel('CRM_HRRecruitment_BAO_HRVacancy', 'location', $locationValue);
+      }
```

Comment:
These 4 lines of code are duplicated in 4 places (twice in this file). Can we have it moved to a helper method somewhere else? I honestly don't know where would be the best place to put it (maybe in CRM_HRRecruitment_BAO_HRVacancy?), but anywhere will be better than having it repeated in a lot of places